fedimint-api/src/encoding/mod.rs

```diff
@@ -81,6 +81,27 @@ impl_encode_decode_tuple!(T1, T2);
 impl_encode_decode_tuple!(T1, T2, T3);
 impl_encode_decode_tuple!(T1, T2, T3, T4);
 
+macro_rules! impl_encode_decode_bool {
+    ($bool_type:ty) => {
+        impl Encodable for $bool_type {
+            fn consensus_encode<W: std::io::Write>(&self, mut writer: W) -> Result<usize, Error> {
+                let bytes = if *self == true { [1u8] } else { [0u8] };
+                writer.write_all(&bytes[..])?;
+                Ok(bytes.len())
+            }
+        }
+        impl Decodable for $bool_type {
+            fn consensus_decode<D: std::io::Read>(mut d: D) -> Result<Self, DecodeError> {
+                let mut bytes = [0u8; 1];
+                d.read_exact(&mut bytes).map_err(DecodeError::from_err)?;
+                Ok(bytes[0] == 1)
+            }
+        }
+    };
+}
```

Comment:
Why use a macro if there's only one implementation?

Reply:
I wanted to take advantage of the existing `derive(Encodable, Decodable)` on `LightningGateway`, reasoning that boolean serialization might be a common need.
That said, we don't need this anyways as we've discussing an approach where we don't save the active flag in DB.